src/twisted/internet/asyncioreactor.py

```diff
@@ -50,6 +73,8 @@ class AsyncioSelectorReactor(PosixReactorBase):
     def __init__(self, eventloop: Optional[SelectorEventLoop] = None):
         if eventloop is None:
             _eventloop: AbstractEventLoop = get_event_loop()
+            if current_async_library_cvar is not None:
+                _eventloop.set_task_factory(factory=sniffioTaskFactory)
```

Comment:
Using this approach we'd have to also wrap  Loop.call_soon(), Loop.call_later(), and Loop.call_at() which also copy the context

Reply:
This strikes me as worth requesting a change so it can't be forgotten?